packages/x-charts/src/BarChart/BarPlot.tsx

```diff
@@ -199,15 +196,45 @@ const useAggregatedData = (): CompletedBarData[] => {
           width: verticalLayout ? barWidth : maxValueCoord - minValueCoord,
           color: colorGetter(dataIndex),
           highlightScope: series[seriesId].highlightScope,
+          value: series[seriesId].data[dataIndex],
+          maskId: `${chartId}_${series[seriesId].stack ?? ''}_${groupIndex}_${dataIndex}`,
```

Comment:
Nitpick: This would produce double underscores (`__`) in case of unstacked data. 🤔 
Maybe you could use `series[seriesId]` or `seriesId` in the "else" case?

Reply:
Nice catch, this sort of has to be consistent, so I went with `stackId || seriesId` since if stack is not present we can use the seriesId without problem instead.